crates/partition-store/src/snapshots/repository.rs

```diff
@@ -152,6 +153,54 @@ impl LatestSnapshot {
     }
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub enum ArchivedLsn {
+    None,
+    Snapshot {
+        // Ordering is intentional: LSN takes priority over elapsed wall clock time for comparisons
+        min_applied_lsn: Lsn,
+        created_at: SystemTime,
+    },
+}
+
+impl ArchivedLsn {
+    pub fn get_min_applied_lsn(&self) -> Lsn {
+        match self {
+            ArchivedLsn::None => Lsn::INVALID,
+            ArchivedLsn::Snapshot {
+                min_applied_lsn, ..
+            } => *min_applied_lsn,
+        }
+    }
+
+    pub fn get_age(&self) -> Result<Duration, SystemTimeError> {
+        match self {
+            ArchivedLsn::None => Ok(Duration::MAX),
+            ArchivedLsn::Snapshot { created_at, .. } => {
+                SystemTime::now().duration_since(*created_at)
```

Comment:
Is it important that we distinguish between created_at > now and created_at == now? If not, then we might get rid fo the error case.

Reply:
No, collapsed to zero - good call to remove the error!